rust/otap-dataflow/crates/otap/src/attributes_processor.rs

```diff
@@ -282,6 +285,11 @@ impl local::Processor<OtapPdata> for AttributesProcessor {
         msg: Message<OtapPdata>,
         effect_handler: &mut local::EffectHandler<OtapPdata>,
     ) -> Result<(), EngineError> {
+        // Start periodic telemetry collection
+        let _ = effect_handler
+            .start_periodic_telemetry(Duration::from_secs(1))
+            .await?;
+
```

Comment:
The `process` method is called for every PData message, so this statement schedules a recurring timer for each message. There is definitely an issue in the current `Processor` interface (and to a lesser extent in the `Receiver`/`Exporter` interfaces) that makes the scheduling of this timer error-prone. I’ll work on it as soon as possible.
In the meantime, I think we should introduce a flag to prevent scheduling a new recurring timer for every received message.